### PR TITLE
Fix DashboardModule.createMenuLink

### PR DESCRIPTION
**Pull request by adrw**
Links which included `/` would fail on page load.

This PR also moves further up in the lifecycle slug and URL path prefix validation for better DX (failures will happen on boot, not on page load).

### CODE REVIEW

File: misk-jdbc-testing/src/main/kotlin/misk/jdbc/ExtendedQueryExecutionListener.kt [hunk 1]
```diff
@@ -47,7 +47,7 @@ open class ExtendedQueryExecutionListener : QueryExecutionListener, MethodExecut
     if (queryInfoList == null) return
 
     for (info in queryInfoList) {
-      val query = info.query.toLowerCase()
+      val query = info.query.lowercase()
```
Comment (adrw) [inline misk-jdbc-testing/src/main/kotlin/misk/jdbc/ExtendedQueryExecutionListener.kt:50]: `.toLowerCase()` is deprecated, `.lowercase()` is the replacement API

File: misk-admin/api/misk-admin.api [hunk 2]
```diff
@@ -339,7 +339,7 @@ public final class misk/web/dashboard/DashboardTabLoaderEntry {
 	public fun toString ()Ljava/lang/String;
 }
 
-public final class misk/web/dashboard/DashboardTabProvider : javax/inject/Provider {
+public final class misk/web/dashboard/DashboardTabProvider : misk/web/dashboard/ValidWebEntry, javax/inject/Provider {
```
Comment (lkerford) [inline misk-admin/api/misk-admin.api:342]: Would you mind explaining how this solve the issue? I'm just not familiar enough with `ValidWebEntry` . The PR looks good but I'm a bit uncertain how it resolved our weird spinner issue

Comment (adrw) [reply misk-admin/api/misk-admin.api:342]: `ValidWebEntry` enforces that the `DashboardTabProvider` has been constructed with a valid slug and URL path prefix. This currently happens later in the lifecycle when the  `DashboardTabProvider` creates the `DashboardTab` on `get()` but that error is opaque and was difficult to debug.

Comment (adrw) [reply misk-admin/api/misk-admin.api:342]: The weird spinner issue was from an endpoint failing with the above invalid slug error. The change to the `DashboardModule` fixed the `createMenuLink` to ensure that the slug is always valid.

**Review by ean5533 — APPROVED** [review]

**Pull request merged**